stac_ipyleaflet/stac_discovery/stac.py

```diff
@@ -1,59 +1,77 @@
 # [SOME] code taken from https://github.com/giswqs/leafmap/blob/master/leafmap/stac.py
 from pystac_client import ItemSearch
-import requests
+from stac_ipyleaflet.stac_discovery.requests import make_get_request
+from stac_ipyleaflet.constants import RESCALE
+from typing import TypedDict, Optional
+import logging
+
+class OutputCollectionObj(TypedDict):
+    id: str
+    title: str
+    start_date: str
+    end_date: str
+    bbox: str
+    metadata: Optional[str]
+    href: Optional[str]
+    description: str
+    license: str
```

Comment:
📓 : Adding this typedDict because it makes for better readability, in-code documentation and type guards. We can start using [pythons typing library](https://docs.python.org/3/library/typing.html) too in our codebase BUT if we arn't resusing these types in mulitple places in the codebase, it might not be as* useful